src/pages/Controllers/Controllers.js

```diff
@@ -90,15 +91,26 @@ function Details() {
     </span>
   );
 
+  function generatePathValue(c) {
+    const column = { content: "" };
+    column.content = c.path === "admin/jaas" ? "JAAS" : c.path;
+    if (!column.content) {
+      column.content = c.wsControllerURL;
+      column.className = "disconnected-controller";
```

Comment:
I see how you got here but setting a status of the connection here would be easier to reason here and leave the presentation of the controller states to the render function or presentational function so the concerns are separated.

Reply:
This function is the function that determines what gets presented as it's called by the table generation code. I could add a `connected` value further up in the stack but I don't want to keep two sources of truth around the connected status. One being the actual connection object and this being the other. In the future these two will be merged so it will be less of an issue. But anyways, for now I think this is the best place for this logic.